server/src/main/java/org/elasticsearch/search/query/QueryPhaseCollectorManager.java

```diff
@@ -238,6 +236,8 @@ static CollectorManager<Collector, QueryPhaseResult> createQueryPhaseCollectorMa
             int trackTotalHitsUpTo = searchContext.scrollContext().totalHits != null
                 ? SearchContext.TRACK_TOTAL_HITS_DISABLED
                 : SearchContext.TRACK_TOTAL_HITS_ACCURATE;
+            // top collectors don't like a size of 0
```

Comment:
NIT: could unwrap the `else` block here so you don't have to duplicate the calculation + comment :) 

Since you return for size 0 above anyway, this can just be 

```
if (searchContext.scrollContext() != null) {

} else {

}
```

without nesting :)

Reply:
good point! Thanks for looking at the code more carefully than I did :)

I flipped a couple of conditionals around to do if (a == null)  rather than the opposite while I was at it. I guess that will make the diff hard to follow.